apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/TracesResourceTest.java

```diff
@@ -4225,6 +4219,60 @@ void feedback__whenFeedbackTraceIdIsNotValid__thenReturn400() {
         }
     }
 
+    @Nested
+    @DisplayName("Opik usage:")
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    class Usage {
+
+        private final String okApikey = UUID.randomUUID().toString();
+
+        @BeforeEach
+        void setUp() {
+            wireMock.server().stubFor(
+                    post(urlPathEqualTo("/opik/auth"))
+                            .withHeader(HttpHeaders.AUTHORIZATION, equalTo(""))
+                            .withRequestBody(matchingJsonPath("$.workspaceName", matching(".+")))
+                            .willReturn(WireMock.unauthorized()));
+        }
+
+        @Test
+        @DisplayName("Get traces count on previous day for all workspaces, no Auth")
+        void tracesCountForWorkspace() {
+            // Setup mock workspace with traces
+            var workspaceName = UUID.randomUUID().toString();
+            var workspaceId = UUID.randomUUID().toString();
+            int tracesCount = setupTracesForWorkspace(workspaceName, workspaceId, okApikey);
+
+            // Change created_at to the previous day in order to capture those traces in count query, since for Stripe we need to count it daily for yesterday
+            String updateCreatedAt = "ALTER TABLE traces UPDATE created_at = subtractDays(created_at, 1) WHERE workspace_id=:workspace_id;";
```

Comment:
This creates side effects for any other test in the system. We should avoid this.

Honestly, there's probably no way to test the logic of this endpoint from an integration test, given their current set up.

I'd propose to just have a basic sanity call in this test where you just assure that the call is successful and you receive some payload. This is something similar to what happens to `com.comet.opik.api.resources.v1.priv.ExperimentsResourceTest.FindExperiments#findAll` 

The inner logic has to be tested manually and with automated tests at some other level (unit or DAO integ tests).

Reply:
I don't thing it creates side effects. I alter data only for one workspace which is used in the test, and workspace id is random UUID generated on the fly. So other tests should be fine, considering that each test uses it's own unique workspace Id.

And I would prefer to have valid test with data, which executes all the logic at least for happy scenario. Not empty sanity check, which is almost no use.

In general we might thing not to run in parallel different tests against same DB, since for example in case we would like to retrieve all storage size (and it will be for all workspaces), it would not be possible in this scenario. 

So I would suggest to have some tests which run sequentially and restore DB to initial state after each one.

Reply:
It's possible to create a test using a dedicated container isolated from the other tests. This would address the concerns.

Reply:
My bad here as I missed that the query is filtering by workspace and that's randomly generated.

Ignore my comment here, the testing strategy should be good in this case.